src/sqlfluff/core/config.py

```diff
@@ -549,12 +565,37 @@ def _validate_configs(
                 else:
                     # Raise an error.
                     raise SQLFluffUserError(
-                        f"Config file {file_path} set an outdated config "
+                        f"Config file {file_path!r} set an outdated config "
                         f"value {formatted_key}.\n\n{removed_option.warning}\n\n"
                         "See https://docs.sqlfluff.com/en/stable/configuration.html"
                         " for more details."
                     )
 
+            # Second validate any layout configs for validity.
+            # NOTE: For now we don't check that the "type" is a valid one
+            # to reference, or that the values are valid. For the values,
+            # these are likely to be rejected by the layout routines at
+            # runtime. The last risk area is validating that the type is
+            # a valid one.
+            if k[0] == "layout":
```

Comment:
Pardon the ignorant question, but do we need to check that k has a length?

```suggestion
            if k and k[0] == "layout":
```

Reply:
We probably don't explicitly - but that does seem sensible 🤔